{
  "uid" : "5bf910dc925aac1b",
  "name" : "test_run[run_case0]",
  "fullName" : "testcase.TestKeywords.TestKeyword#test_run[run_case0]",
  "historyId" : "1b1458eb6c9578bbae1b78493d33e22c",
  "time" : {
    "start" : 1588773493222,
    "stop" : 1588773493222,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "AttributeError: 'NoneType' object has no attribute 'launch_app'",
  "statusTrace" : "self = <testcase.TestKeywords.TestKeyword object at 0x000001FC304E4E20>\nstart_appium_desired = None\nrun_case = {'CASE_ID': 'Case_001', 'TC_ID': 'TC_Login', '备注': '登录功能测试', '描述': '登录功能测试', ...}\n\n    @pytest.mark.parametrize(\"run_case\", run_list)\n    def test_run(self, start_appium_desired, run_case):\n        self.driver = start_appium_desired\n>       self.driver.launch_app()\nE       AttributeError: 'NoneType' object has no attribute 'launch_app'\n\ntestcase\\TestKeywords.py:29: AttributeError",
  "flaky" : false,
  "beforeStages" : [ {
    "name" : "cmdopt",
    "time" : {
      "start" : 1588773491208,
      "stop" : 1588773491208,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : false,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  }, {
    "name" : "start_appium_desired",
    "time" : {
      "start" : 1588773491208,
      "stop" : 1588773493221,
      "duration" : 2013
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : false,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  } ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "AttributeError: 'NoneType' object has no attribute 'launch_app'",
    "statusTrace" : "self = <testcase.TestKeywords.TestKeyword object at 0x000001FC304E4E20>\nstart_appium_desired = None\nrun_case = {'CASE_ID': 'Case_001', 'TC_ID': 'TC_Login', '备注': '登录功能测试', '描述': '登录功能测试', ...}\n\n    @pytest.mark.parametrize(\"run_case\", run_list)\n    def test_run(self, start_appium_desired, run_case):\n        self.driver = start_appium_desired\n>       self.driver.launch_app()\nE       AttributeError: 'NoneType' object has no attribute 'launch_app'\n\ntestcase\\TestKeywords.py:29: AttributeError",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "5939abdc4a796c50",
      "name" : "log",
      "source" : "5939abdc4a796c50.txt",
      "type" : "text/plain",
      "size" : 61
    } ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : true,
    "shouldDisplayMessage" : true,
    "attachmentsCount" : 1
  },
  "afterStages" : [ {
    "name" : "start_appium_desired::0",
    "time" : {
      "start" : 1588773493332,
      "stop" : 1588773493333,
      "duration" : 1
    },
    "status" : "broken",
    "statusMessage" : "AttributeError: 'NoneType' object has no attribute 'quit'\n",
    "statusTrace" : "  File \"C:\\Users\\cjj\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\allure_commons\\_allure.py\", line 169, in __call__\n    return self._fixture_function(*args, **kwargs)\n  File \"C:\\Users\\cjj\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\_pytest\\fixtures.py\", line 802, in _teardown_yield_fixture\n    next(it)\n  File \"C:\\Users\\cjj\\PycharmProjects\\Appium_AutoTest\\testcase\\conftest.py\", line 42, in start_appium_desired\n    driver.quit()\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : true,
    "shouldDisplayMessage" : true,
    "attachmentsCount" : 0
  } ],
  "labels" : [ {
    "name" : "parentSuite",
    "value" : "testcase"
  }, {
    "name" : "suite",
    "value" : "TestKeywords"
  }, {
    "name" : "subSuite",
    "value" : "TestKeyword"
  }, {
    "name" : "host",
    "value" : "DESKTOP-0LL74L3"
  }, {
    "name" : "thread",
    "value" : "2212-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "testcase.TestKeywords"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "run_case",
    "value" : "{'TC_ID': 'TC_Login', 'CASE_ID': 'Case_001', '是否运行': 'y', '用例名称': '登录失败', '测试数据': 'username=13718418220,password=123456', '期望结果': '密码错误', '备注': '登录功能测试', '描述': '登录功能测试'}"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "5bf910dc925aac1b.json",
  "parameterValues" : [ "{'TC_ID': 'TC_Login', 'CASE_ID': 'Case_001', '是否运行': 'y', '用例名称': '登录失败', '测试数据': 'username=13718418220,password=123456', '期望结果': '密码错误', '备注': '登录功能测试', '描述': '登录功能测试'}" ]
}